c2rust-analyze/src/c_void_casts.rs

```diff
@@ -245,16 +257,17 @@ impl<'tcx> CVoidCasts<'tcx> {
     /// See [`CVoidCastsUniDirectional::get_adjusted_place_or_default_to`].
     pub fn get_adjusted_place_or_default_to(
         &self,
+        loc: &Location,
         direction: CVoidCastDirection,
         place: Place<'tcx>,
     ) -> Place<'tcx> {
         self.direction(direction)
-            .get_adjusted_place_or_default_to(place)
+            .get_adjusted_place_or_default_to(loc, place)
     }
 
     /// See [`CVoidCastsUniDirectional::insert`].
-    fn insert(&mut self, direction: CVoidCastDirection, cast: CVoidCast<'tcx>) {
-        self.direction_mut(direction).insert(cast)
+    fn insert(&mut self, loc: Location, direction: CVoidCastDirection, cast: CVoidCast<'tcx>) {
```

Comment:
I'd switch `loc` and `direction`.  That's the first lookup before `location`.

Reply:
65d1bce3